src/preprocessing/mod.rs

```diff
@@ -0,0 +1,5 @@
+/// Transform a data matrix by replaceing all categorical variables with their one-hot vector equivalents
+pub mod categorical_encoder;
```

Comment:
The module name is a bit too long. Can you rename it to either `categorical` or `one_hot_encoder`?